content/contributing/helping-out/writing-docs.md

```diff
@@ -0,0 +1,237 @@
+4. Examples must pass Bevy's standard `clippy` lints.
+5. The polish level of your examples should correspond to the point you're trying to make.
+   1. If you're demonstrating a new feature, show only the most basic syntax as locally as possible.
+   2. When trying to explain how a game can be made, organize and polish your code to showcase best practices.
+   3. Lack of polish should serve an end: don't show bad or sloppy practices without a good reason.
+   4. Showing how (and why!) to refactor your code is a very powerful teaching tool.
+6. Stick to a consistent style (e.g. for loops vs map) within each example.
+7. If you need to give advice that will only matter to some of your audience (e.g. how to handle an edge case, or support a specific platform), do so in a clearly marked aside or list.
+8. Examples should not use or rely on third-party plugins.
+These may be appropriate to link in "next steps" however at the end of the examples.
+   1. Third-party crates should be limited to the most essential, such as `rand`.
+9. If additional code block attributes like `no_run` or `hide-lines=x-y` need to be specified, you should always order these so that the language is the last attribute. If we would specify `rust,no_run` the syntax highlighting wouldn't work, but changing it to `no_run,rust` makes it work.
+10. To validate if local code changes are compiling you can `cd` into the `code-validation` folder and test your code using `cargo test`.
+11. To make sure your web-based files (html, markdown) are formatted correctly run the commands:
+
+    ```sh
+    markdownlint -f -c .github/linters/.markdown-lint.yml .
+    djlint
+    typos
+    ```
+
+    in the root directory of your local Bevy website repository. This will format markdown files and tell you the issues in HTML files. In order to run the command you should install `markdownlint-cli`, `djlint`, and `typos-cli`. See: <https://github.com/igorshubovych/markdownlint-cli>, <https://www.djlint.com/docs/getting-started/>, and <https://github.com/crate-ci/typos?tab=readme-ov-file#install>.  Note that the CI also includes `editorconfigchecker` but there isn't an easy way to run this manually, so you should instead rely on CI to validate files with this tool.
+12. To hide lines of code in Zola Rust code blocks of the book you should: a. Mark each line you wish to hide with a `#` with an empty space afterwards like `# //...line_of_code_here...` although you were hiding lines in rustdoc. b. Run the [utility tool](write-rustdoc-hide-lines) in `<ROOT_OF_WEBSITE>/write_rustdoc-hide-lines/` named `write_rustdoc_hide_lines.sh`.
+13. To reference Rust API docs you can use markdown's reference-style links like so:
+   [`HashMap`]
+
+   ```md
+   [`HashMap`]
+
+   [`HashMap`]: https://doc.rust-lang.org/std/collections/struct.HashMap.html
+   ```
+
+   [`HashMap`]: https://doc.rust-lang.org/std/collections/struct.HashMap.html
```

Comment:
```suggestion
## Contributor's style guide

When writing and reviewing learning material for The Bevy Book and The Quick Start Guide games, please try to follow these guidelines:

### Writing

1. Use clear, simple language.
2. Prefer short sentences. Remove extra words.
3. **Bold** new vocabulary words where they are defined.
   1. Define them as soon as is reasonable after they are introduced.
4. Make sure your grammar and spelling are correct.
5. Avoid idioms and slang.
6. Speak directly to the reader in an approachable tone. Use "we" and "you" pronouns.
7. It can be useful to create specific, named characters to demonstrate a point.
   1. If you do, pick a pronoun set for them and stick to it.
   2. Otherwise, use  "they/them" third-person pronouns to refer to the reader or others.
8. Keep humor light.
   1. Avoid off-color or offensive humor.
   2. Be mindful not to overuse in-jokes or cultural references.
   3. Don't drag your jokes out: that's not what the audience is here to read.

### Organizational

1. Carefully organize your work into separate pages, headings, paragraphs and code blocks.
2. Clearly signal when you are explaining a concept in technical depth so it can be skipped.
3. Use lists, numbered lists and sub-lists to present information in bite-sized ways.
   1. Refer back to these items by number!
4. Provide plenty of links, but be sure that what you are linking to is obvious by context.
   1. Link to other sections of the book / example / web page when you mention them.
   2. Always link to the most specific location you can, whether that's a section on a page or a method on a struct.
   3. Use the `latest` tag when linking to Bevy docs and source code so it won't go stale every time the version is updated.
   4. When linking to detailed explanations or discussions, summarize the most important points in addition to providing a link.

### Technical

1. All examples must be able to be compiled and run.
2. Prefer game-relevant, descriptive examples and variable names over generic ones like `MyEvent`. Avoid meaningless names like `foo` at all times.
3. It's good practice to break your code into blocks with comments or explanatory text, but you need to link to a cohesive, copy-able whole at the end.
4. Examples must pass Bevy's standard `clippy` lints.
5. The polish level of your examples should correspond to the point you're trying to make.
   1. If you're demonstrating a new feature, show only the most basic syntax as locally as possible.
   2. When trying to explain how a game can be made, organize and polish your code to showcase best practices.
   3. Lack of polish should serve an end: don't show bad or sloppy practices without a good reason.
   4. Showing how (and why!) to refactor your code is a very powerful teaching tool.
6. Stick to a consistent style (e.g. for loops vs map) within each example.
7. If you need to give advice that will only matter to some of your audience (e.g. how to handle an edge case, or support a specific platform), do so in a clearly marked aside or list.
8. Examples should not use or rely on third-party plugins.
These may be appropriate to link in "next steps" however at the end of the examples.
   1. Third-party crates should be limited to the most essential, such as `rand`.
9. If additional code block attributes like `no_run` or `hide-lines=x-y` need to be specified, you should always order these so that the language is the last attribute. If we would specify `rust,no_run` the syntax highlighting wouldn't work, but changing it to `no_run,rust` makes it work.
10. To validate local code changes you can either `./learning-code-examples/validate_examples.sh` from anywhere, or from the project's root `cd learning-code-examples && cargo check --examples && cargo clippy --examples && cargo fmt --check`.
11. To make sure your web-based files (html, markdown) are formatted correctly run the commands:

    ```
    markdownlint -f -c .github/linters/.markdown-lint.yml .
    djlint
    typos
    ```

    in the root directory of your local Bevy website repository. This will format markdown files and tell you the issues in HTML files. In order to run the command you should install `markdownlint-cli`, `djlint`, and `typos-cli`. See: <https://github.com/igorshubovych/markdownlint-cli>, <https://www.djlint.com/docs/getting-started/>, and <https://github.com/crate-ci/typos?tab=readme-ov-file#install>.  Note that the CI also includes `editorconfigchecker` but there isn't an easy way to run this manually, so you should instead rely on CI to validate files with this tool.
12. To reference Rust API docs you can use markdown's reference-style links like so:
   [`HashMap`]

   ```md
   [`HashMap`]

   [`HashMap`]: https://doc.rust-lang.org/std/collections/struct.HashMap.html
   \```

   [`HashMap`]: https://doc.rust-lang.org/std/collections/struct.HashMap.html
```
Update this part to where it is currently at in the main `CONTRIBUTING.md` of the site. 
> [!NOTE]
> Remove the backslash after this change is made; GitHub was acting weird with the markdown and broke the suggestion unless I added the backslash temporarily.

Reply:
As a fun tip, you can nest code blocks by increasing the amount of backticks:

`````markdown
# Hello, world!

```markdown
Markdown **recursion**!

````rust
println!("Hello, world!");
````
```
`````